mne/channels/tests/test_interpolation.py

```diff
@@ -16,68 +15,95 @@
     """Test interpolation"""
-    epochs.info['bads'] = []
-    goods_idx = np.ones(len(epochs.ch_names), dtype=bool)
-    goods_idx[epochs.ch_names.index('EEG 012')] = False
+    raw, epochs, epochs_eeg, epochs_meg = _load_data()
+
+    # create good and bad channels for EEG
+    epochs_eeg.info['bads'] = []
+    goods_idx = np.ones(len(epochs_eeg.ch_names), dtype=bool)
+    goods_idx[epochs_eeg.ch_names.index('EEG 012')] = False
     bads_idx = ~goods_idx
 
-    evoked = epochs.average()
-    ave_before = evoked.data[bads_idx]
+    evoked_eeg = epochs_eeg.average()
+    ave_before = evoked_eeg.data[bads_idx]
 
-    pos = epochs._get_channel_positions()
+    # interpolate bad channels for EEG
+    pos = epochs_eeg._get_channel_positions()
     pos_good = pos[goods_idx]
     pos_bad = pos[bads_idx]
     interpolation = _make_interpolation_matrix(pos_good, pos_bad)
-    assert_equal(interpolation.shape, (1, len(epochs.ch_names) - 1))
+    assert_equal(interpolation.shape, (1, len(epochs_eeg.ch_names) - 1))
+    ave_after = np.dot(interpolation, evoked_eeg.data[goods_idx])
 
-    ave_after = np.dot(interpolation, evoked.data[goods_idx])
+    epochs_eeg.info['bads'] = ['EEG 012']
+    evoked_eeg = epochs_eeg.average()
+    assert_array_equal(ave_after, evoked_eeg.interpolate_bads().data[bads_idx])
 
     assert_allclose(ave_before, ave_after, atol=2e-6)
 
-    epochs.info['bads'] = []
-    assert_raises(ValueError, epochs.interpolate_bads_eeg)
-
-    epochs.info['bads'] = ['EEG 012']
-    epochs.preload = False
-    assert_raises(ValueError,  epochs.interpolate_bads_eeg)
-    epochs.preload = True
-
-    epochs2.info['bads'] = ['EEG 012', 'MEG 1711']
+    # check that interpolation fails when preload is False
+    epochs_eeg.preload = False
+    assert_raises(ValueError,  epochs_eeg.copy().interpolate_bads)
+    epochs_eeg.preload = True
 
-    epochs2.interpolate_bads_eeg()
-    ave_after2 = epochs2.average().data[n_meg + np.where(bads_idx)[0]]
-
-    assert_array_almost_equal(ave_after, ave_after2, decimal=16)
-
-    raw = io.RawArray(data=epochs._data[0], info=epochs.info)
-    raw_before = raw._data[bads_idx]
-    raw.interpolate_bads_eeg()
-    raw_after = raw._data[bads_idx]
+    # check that interpolation changes the data in raw
+    raw_eeg = io.RawArray(data=epochs_eeg._data[0], info=epochs_eeg.info)
+    raw_before = raw_eeg._data[bads_idx]
+    raw_after = raw_eeg.interpolate_bads().copy()._data[bads_idx]
     assert_equal(np.all(raw_before == raw_after), False)
 
-    evoked = epochs.average()
-    evoked.interpolate_bads_eeg()
-    assert_array_equal(ave_after, evoked.data[bads_idx])
-
+    # check that interpolation fails when preload is False
     for inst in [raw, epochs]:
         assert hasattr(inst, 'preload')
         inst.preload = False
         inst.info['bads'] = [inst.ch_names[1]]
-        assert_raises(ValueError, inst.interpolate_bads_eeg)
+        assert_raises(ValueError, inst.copy().interpolate_bads)
```

Comment:
that is really confusing, it might be indicative of some underlying pathological behavior, e.g. you start modifying the object before the `ValueError` is caught such that the next call does something you don't expect